neuro_san/service/generic/async_agent_service.py

```diff
@@ -208,7 +208,9 @@ async def streaming_chat(self, request_dict: Dict[str, Any],
 
         # Prepare
         factory = ExternalAgentSessionFactory(use_direct=False)
-        invocation_context = SessionInvocationContext(factory, self.llm_factory, self.toolbox_factory, metadata)
+        invocation_context = SessionInvocationContext(factory, self.llm_factory,
+                                                      self.toolbox_factory, metadata,
+                                                      originates_from_async=True)
```

Comment:
Tell the SessionInvocationContext that we are originating from an asynchronous source.